pkg/secrets/fetch_secret.go

```diff
@@ -49,6 +54,35 @@ func execCommand(inputPayload string) ([]byte, error) {
 	defer cancel()
 
 	cmd := exec.CommandContext(ctx, secretBackendCommand, secretBackendArguments...)
+
+	if len(secretBackendCommandSHA256) != 0 {
+		if !filepath.IsAbs(secretBackendCommand) {
+			return nil, fmt.Errorf("error while running '%s': absolute path required with SHA256", secretBackendCommand)
+		}
+
+		if err := checkConfigRights(configFileUsed); err != nil {
+			return nil, err
+		}
+
+		f, err := lockOpenFile(secretBackendCommand)
+		if err != nil {
+			return nil, err
+		}
+		defer f.Close() //nolint:errcheck
+
+		sha256, err := fileHashSHA256(f)
+		if err != nil {
+			return nil, err
+		}
+
+		if !strings.EqualFold(sha256, secretBackendCommandSHA256) {
+			return nil, fmt.Errorf("error while running '%s': SHA256 mismatch", secretBackendCommand)
```

Comment:
nit: It would be good here to print both the expected and actual SHAs for easier debugging.